apps/pdal.cpp

```diff
@@ -208,17 +215,76 @@ void App::addArgs(ProgramArgs& args)
         m_showOptions);
 }
 
+namespace
+{
```

Comment:
You can't convert a stateful lambda to a function pointer, so I had to hoist it to an accessible scope.  The rename to `logPtr` was to avoid possible conflicts of the name `log` with `cmath`.